### PR TITLE
Various ag grid theme fixes

### PR DESCRIPTION
**Pull request by origami-z**
- [x] Adds custom dropdown editor example
- [x] Adds custom flag renderer example
- [x] Ag Menu padding to match Salt menu (0 padding) 
- [x] Group cell line height mismatch
- [x] Custom cell editor background color when focused (should be black in dark, instead of blue)
- [x] Double focus ring for a custom dropdown editor
- Changed base size implementation in HD Compact

#3775, #3675

Design related task to support HD compact in Figma Next: https://github.com/jpmorganchase/salt-ds/issues/3850

### CODE REVIEW

File: .changeset/large-carrots-end.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+"@salt-ds/ag-grid-theme": patch
+---
+
+- Fixed background color for custom editor component.
+- Fixed header text being cropped in HD compact. Closes #3675.
+- Fixed Country Symbol taller than expected in HD compact. This alters `--salt-size-base` token so Salt Button, form controls (Input, Dropdown, Combo Box) will be impacted as well. Closes #3775.
+- Fixed group value not center aligned vertically.
+- Updated ag grid menu styling to match closer to Salt Menu component.
+
+Note: We previously made a mistake on `rowHeight` recommendation when configurating AG Grid, which should be 1px more to account for border between row.
```
Comment (origami-z) [inline .changeset/large-carrots-end.md:11]: FYI @bhoppers2008

File: packages/ag-grid-theme/src/dependencies/cell-editors/DropdownEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        props.api!.stopEditing();
+      }, 100);
+    },
+    [props.api],
+  );
+
+  const onEscapeKeyPressed = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === "Escape") {
+        props.api!.stopEditing();
```
Comment (joshwooding) [inline packages/ag-grid-theme/src/dependencies/cell-editors/DropdownEditor.tsx:43]:
```suggestion
        props.api?.stopEditing();
```

File: packages/ag-grid-theme/src/dependencies/cell-editors/DropdownEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+  const onSelect = useCallback(
+    (_event: SyntheticEvent, selectedValue: Array<GridCellValue>) => {
+      setValue(selectedValue[0]);
+
+      // hack to make selection actually record the edit.
+      // timeout is necessary because otherwise the grid stops editing
+      // before the useImperativeHandle getValue returns the new value state
+      setTimeout(() => {
+        props.api!.stopEditing();
```
Comment (joshwooding) [inline packages/ag-grid-theme/src/dependencies/cell-editors/DropdownEditor.tsx:34]:
```suggestion
        props.api?.stopEditing();
```

File: packages/ag-grid-theme/src/dependencies/cell-editors/DropdownEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+export interface DropdownEditorParams extends ICellEditorParams {
+  source?: Array<GridCellValue>;
+  dropdownProps?: Partial<DropdownProps<string>>;
+}
+
+export const DropdownEditor = forwardRef((props: DropdownEditorParams, ref) => {
+  const { value: initialValue, source = [], dropdownProps } = props;
+  const [value, setValue] = useState(initialValue);
+
+  const button = React.useRef<HTMLButtonElement>(null);
```
Comment (joshwooding) [inline packages/ag-grid-theme/src/dependencies/cell-editors/DropdownEditor.tsx:24]:
```suggestion
  const button = useRef<HTMLButtonElement>(null);
```
?

File: packages/ag-grid-theme/src/dependencies/cell-editors/DropdownEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    (e: KeyboardEvent) => {
+      if (e.key === "Escape") {
+        props.api!.stopEditing();
+      }
+    },
+    [props.api],
+  );
+
+  useEffect(() => {
+    button.current!.focus();
```
Comment (joshwooding) [inline packages/ag-grid-theme/src/dependencies/cell-editors/DropdownEditor.tsx:50]:
```suggestion
    button.current?.focus();
```

**Review by mark-tate — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**